internal/renderer/service_util.go

```diff
@@ -341,6 +343,51 @@ func createLbService4Gateway(c *RenderContext, gw *gwapiv1a2.Gateway) *corev1.Se
 		})
 	}
 
+	// Set health check port
+	regexProtocol := `^service\.beta\.kubernetes\.io\/.*health.*protocol$`
+	regexPort := `^service\.beta\.kubernetes\.io\/.*health.*port$`
+
+	annotationRegexProtocol := regexp.MustCompile(regexProtocol)
+	annotationRegexPort := regexp.MustCompile(regexPort)
+
+	var healthCheckPort int32
+	healthCheckProtocol := "TCP"
+
+	// Find health check port
+	for annotationKey, annotation := range c.gwConf.Spec.LoadBalancerServiceAnnotations {
+		match := annotationRegexPort.FindStringSubmatch(annotationKey)
```

Comment:
I'd recommend using [`MatchString`](https://pkg.go.dev/regexp#Regexp.MatchString) here, it's much simpler.